src/test/java/mate/academy/bookstore/repository/shoppingcart/ShoppingCartRepositoryTest.java

```diff
@@ -0,0 +1,96 @@
+package mate.academy.bookstore.repository.shoppingcart;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.util.Optional;
+import java.util.Set;
+import javax.sql.DataSource;
+import lombok.SneakyThrows;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ShoppingCartRepositoryTest {
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+        setup(dataSource);
+    }
+
+    @Test
+    @DisplayName("Verify findByUserId() method works for ShoppingCartRepository")
+    void findByUserId_ValidUserId_returnOptionalShoppingCart() {
+        ShoppingCart expected = shoppingCartFromDb();
+        Optional<ShoppingCart> actual = shoppingCartRepository.findByUserId(17L);
+        System.out.println(expected.getCartItems());
```

Comment:
Is it possible to divide test into 3 logical parts (precondition, action and assertion) by white spaces?